File: src/types/apiKey.ts
export interface ApiKey {
  id: number
  provider_id: number
  name: string
  masked_preview: string
  parent_id: number | null
  created_at: string
  updated_at: string
}

export interface ApiKeyView {
  id: number
  provider_id: number
  provider_name: string
  provider_display_name: string
  name: string
  masked_preview: string
  parent_id: number | null
  created_at: string
  updated_at: string
}

export interface CreateKeyInput {
  provider_id: number
  name: string
  raw_key: string
  parent_id?: number
}

export interface UpdateKeyInput {
  id: number
  provider_id: number
  name: string
  raw_key?: string
  parent_id?: number
}

File: src-tauri/src/models/api_key.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ApiKey {
    pub id: i64,
    pub provider_id: i64,
    pub name: String,
    pub encrypted_key: Vec<u8>,
    pub iv: Vec<u8>,
    pub masked_preview: String,
    pub parent_id: Option<i64>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ApiKeyView {
    pub id: i64,
    pub provider_id: i64,
    pub provider_name: String,
    pub provider_display_name: String,
    pub name: String,
    pub masked_preview: String,
    pub parent_id: Option<i64>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct NewApiKey {
    pub provider_id: i64,
    pub name: String,
    pub raw_key: String,
    pub parent_id: Option<i64>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UpdateApiKey {
    pub id: i64,
    pub provider_id: i64,
    pub name: String,
    pub raw_key: Option<String>,
    pub parent_id: Option<i64>,
}
